lib/cli/interactive-setup/aws-credentials.js

```diff
@@ -86,33 +193,208 @@ const steps = {
+        return;
+      }
+      throw err;
+    }
+    try {
+      await sdk.createProviderLink(orgUid, linkType, linkUid, providerUid);
+    } catch (err) {
+      if (err.statusCode && err.statusCode >= 500) {
+        process.stdout.write(
+          `\n${chalk.yellow(
+            'Dashboard service is currently unavailable, please try again later'
+          )}\n`
+        );
+        return;
+      }
+      throw err;
+    }
+  },
 };
 
 module.exports = {
-  async isApplicable({ configuration }) {
+  async isApplicable(context) {
+    const { configuration } = context;
     if (
       _.get(configuration, 'provider') !== 'aws' &&
       _.get(configuration, 'provider.name') !== 'aws'
     ) {
       return false;
     }
     if (new AWS.S3().config.credentials) return false;
-    return !(await awsCredentials.resolveFileProfiles()).size;
+    if ((await awsCredentials.resolveFileProfiles()).size) return false;
+
+    const orgName = configuration.org;
+    if (orgName) {
+      let providers;
+      try {
+        providers = await getProviders(orgName);
+      } catch (err) {
+        if (err.code === 'DASHBOARD_UNAVAILABLE') {
+          process.stdout.write(`\n${chalk.yellow(err.message)}\n`);
+          return false;
+        }
+        throw err;
+      }
+      const hasDefaultProvider = providers.some((provider) => provider.isDefault);
+
+      if (hasDefaultProvider) return false;
+    }
+
+    return true;
   },
   async run(data) {
-    process.stdout.write(
-      'No AWS credentials were found on your computer, ' +
-        'you need these to host your application.\n\n'
-    );
-
     return module.exports.runSteps(data);
   },
   steps,
   runSteps: async (context) => {
-    if (!(await steps.shouldSetupAwsCredentials())) return;
-    await steps.ensureAwsAccount();
+    let providers;
+
+    // It is possible that user decides to not configure org for his service and
+    // we still should allow setup of local credentials in such case
+    if (context.configuration.org) {
+      try {
+        providers = await getProviders(context.configuration.org);
+      } catch (err) {
+        if (err.code === 'DASHBOARD_UNAVAILABLE') {
+          process.stdout.write(`\n${chalk.yellow(err.message)}\n`);
+          return;
+        }
+        throw err;
+      }
+    }
+    const credentialsSetupChoiceAnswer = await credentialsSetupChoice(providers);
+
+    if (credentialsSetupChoiceAnswer === CREDENTIALS_SETUP_CHOICE.CREATE_PROVIDER) {
+      try {
+        const createdProviderUid = await steps.handleProviderCreation(context);
+        const hadExistingProviders = Boolean(providers.length);
+        if (createdProviderUid && hadExistingProviders) {
+          // This is situation where user decided to create a new provider and already had previous providers setup
+          // In this case, we want to setup an explicit link between provider and service as the newly created provider
+          // might not be the default one.
```

Comment:
It is possible that the user created a default one in such case, however, an explicit link in such a situation will ensure that we don't need to check for that. We talked about it during the call and the decision was to "just" explicitly link in situations where there already were existing providers.

Reply:
I see a difference between falling back to default, and hard assigning a value that's currently default. e.g. imagining scenario when user created default. What will happen if afterwards other provider is switched to be default. Will service now fallback to this new default? Or will it stick to previous provider that's no longer a default (?)

Just want to be sure it's acknowledged. I understand that it might be desirable to hard assign a default created currently (as it was created for this specific service)

Reply:
> What will happen if afterwards other provider is switched to be default. Will service now fallback to this new default? Or will it stick to previous provider that's no longer a default (?)

In such case, it will not fallback to the new default as it has a "hard" link created. We discussed it deeply during one of the last meetings and the conclusion was that:
a) This is not going to be very common flow - in most cases the provider created via CLI flow will be the first one which will always be the default one
b) If user already has other providers and don't have a default one, they probably have a very good reason to have it setup this way (if there's default one it will be used automatically without explicit link) and during creation of a new one they will have to explicitly opt-in to make the new provider the default one (which is different than in case there's 0 currently defined providers)

In all, this flow is not ideal as the "assignment" is based on a best-effort basis (the user on the Dashboard side doesn't really "select" the provider but rather we detect the creation of a new provider and assume it's the one they want to use) - I believe ideally it would be setup in a way that after creation they have an option to select a provider explicitly, but it would be harder to implement.

Reply:
We reiterated offline and agreed that it's fine to hard assign the provider even it was set to be default.

Provider is created in context of this service, so it seems logical to hard code it to it (although line is blurry, as in case of first ever provider we won't hard-code it, with assumption that user created general provider for all services)